CoreFoundation/PlugIn.subproj/CFBundle_Executable.c

```diff
@@ -303,38 +275,96 @@ static CFURLRef _CFBundleCopyExecutableURLInDirectory2(CFBundleRef bundle, CFURL
     return executableURL;
 }
 
+static CFStringRef _CFStringRemovePrefixSuffix(CFStringRef str, CFStringRef prefix, CFStringRef suffix) {
+    CFMutableStringRef mutableString = CFStringCreateMutableCopy(kCFAllocatorSystemDefault, 0, str);
+    
+    // Check if the executable has shared library prefix/suffix.
+    CFRange prefixRange = CFStringFind(mutableString,
+                                       _CFBundleSharedLibraryFilenamePrefix,
+                                       kCFCompareAnchored);
+    CFRange suffixRange = CFStringFind(mutableString,
+                                       _CFBundleSharedLibraryFilenameSuffix,
+                                       kCFCompareAnchored|kCFCompareBackwards);
+    
+    // Only return the stripped string if both the prefix and suffix are found.
+    if (prefixRange.location != kCFNotFound &&
+        suffixRange.location != kCFNotFound &&
+        prefixRange.location == 0 &&
+        suffixRange.location + suffixRange.length == CFStringGetLength(mutableString)) {
+        CFStringReplace(mutableString, prefixRange, CFSTR(""));
+        
+        suffixRange = CFStringFind(mutableString,
+                                   _CFBundleSharedLibraryFilenameSuffix,
+                                   kCFCompareAnchored|kCFCompareBackwards);
+        
+        CFStringReplace(mutableString, suffixRange, CFSTR(""));
+        return mutableString;
+    }
+    else {
+        return NULL;
+    }
+}
+
 static CFURLRef _CFBundleCopyBundleURLForExecutablePath(CFStringRef str) {
-    //!!! need to handle frameworks, NT; need to integrate with NSBundle - drd
-    UniChar buff[CFMaxPathSize];
-    CFIndex buffLen;
-    CFURLRef url = NULL;
-    CFStringRef outstr;
+    // Needs to handle:
+    // - Classic NEXTStep-style bundles (`.framework`, `.app`, `.appex`)
+    // - Freestanding bundles (`Bundle.resource` as sibling of the library/executable)
+    // - FHS bundles (`prefix/share/Bundle.resource`)
+    //
+    // Note:
+    // For freestanding and FHS bundles, this function has to support removing library/executable prefix/suffix
+    // on each platform. For example, it has to detect to the bundle name as `Bundle` for executables
+    // like `Bundle.exe` in Windows or `libBundle.so` on Linux.
     
-    buffLen = CFStringGetLength(str);
-    if (buffLen > CFMaxPathSize) buffLen = CFMaxPathSize;
-    CFStringGetCharacters(str, CFRangeMake(0, buffLen), buff);
+    CFURLRef url = NULL;
     
-#if DEPLOYMENT_TARGET_WINDOWS
-    // Is this a .dll or .exe?
-    if (buffLen >= 5 && (_wcsnicmp((wchar_t *)&(buff[buffLen-4]), L".dll", 4) == 0 || _wcsnicmp((wchar_t *)&(buff[buffLen-4]), L".exe", 4) == 0)) {
-        CFIndex extensionLength = CFStringGetLength(_CFBundleSiblingResourceDirectoryExtension);
-        buffLen -= 4;
-        // If this is an _debug, we should strip that before looking for the bundle
-        if (buffLen >= 7 && (_wcsnicmp((wchar_t *)&buff[buffLen-6], L"_debug", 6) == 0)) buffLen -= 6;
```

Comment:
It looks like support for `_debug` may have been lost in this code and above?

Reply:
Is this longer required by Core Foundation on Windows?